source/Core/Threads/OperatingModes/Soldering.cpp

```diff
@@ -23,14 +23,50 @@ void gui_solderingMode(uint8_t jumpToSleep) {
   bool buttonsLocked = false;
   currentMode        = OperatingMode::soldering;
 
+#ifdef PROFILE_SUPPORT
```

Comment:
It would be better to split out the profile based logic into its own `.cpp` file for readability; rather than the large `#ifdef` blocks if possible.

Reply:
I thought about doing that, but it would lead to lots of duplicate code. Not sure how good that would be?

Reply:
It's certainly a tradeoff.

Given at the moment that the only device with profile support has a lot more flash (2x) compared to the iron's from Miniware; being able to ensure its fully cut out of the compile is probably a win.

But also my main concern with this is that making changes gets more messy when we effectively have two different paths interleaved.

Could try and extract the common parts out to inline'd functions to get somewhat close to how it is now?

Reply:
Done.